views/index.erb

```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title> Chitter </title>
+  </head>
+  <body>
+    <header>
+      <h1>Welcome to Chitter: <small>The Completely Original Website</small></h1>
+    </header>
+    <%if @user %>
+      <h3> Welcome, <%=@user.username%> </h3> 
+    <% end %>
+    <h3 style="color:red" id="flash">
+      <%= flash[:notice] %>
+    </h3>
+    <br>
+    <br>
+    <% if !@user %>
+      <form action="/sign-up" method="get">
+        <input type="submit" value="Log in">
+        <input type="submit" value="Sign up">
+      </form>
+      <br>
+      <br>
+    <% else %> 
+      <form action="/logging-out" method="post">
+        <input type="submit" value="Log out">
+      </form>
+      <br>
+      <br>
+      <form action="/add_post" method="post">
+        <label for="add_post">
+        Create New Post
+          <input type="text" name="peep">
+        </label>
+        <input type="submit" value="Send Peep">
+      </form>
+      <br>
+      <br>
+    <% end %>
+      <h3> Your Chitter Feed </h3>
+    <ul>
+      <% @post.each do |peep| %>
+      <%time = peep[1].strftime("%d/%m/%Y %k:%M")%>
```

Comment:
AR generates "on the fly" attribute accessors for the model class, depending on the column names it finds in the table — which means you can use `peep.id`, `peep.created_at`, `peep.peep`, etc, rather than accessing it like an array index, so it makes the code easier to read

Reply:
That's really good to know, I took ages looking for how to do it the way I did, good to know that I can sort this in a much more manageable way.